### PR TITLE
Add metrics to dynamic worker pool scheduler

### PR DESCRIPTION
**Pull request by rodrigozhou**
## What changed?

Add metrics to dynamic worker pool scheduler:
- buffer size
- active workers

### CODE REVIEW

File: common/tasks/dynamic_worker_pool_scheduler.go [hunk 1]
```diff
@@ -100,12 +119,16 @@ func (pool *DynamicWorkerPoolScheduler) TrySubmit(task Runnable) bool {
 	pool.mu.Lock()
 	if pool.runningGoroutines >= pool.limiter.Concurrency() {
 		enqueued := pool.tryEnqueueLocked(task)
+		bufferSize := pool.bufferSize
 		pool.mu.Unlock()
+		pool.recordBufferSize(bufferSize)
```
Comment (bergundy) [inline common/tasks/dynamic_worker_pool_scheduler.go:124]: We discussed having only the background goroutine record gauges and having separate counters for enqueues/dequeues/buffered/rejected tasks on this path.

File: service/history/outbound_queue_standby_task_executor_test.go
Comment (bergundy) [inline service/history/outbound_queue_standby_task_executor_test.go:1]: This doesn't belong in this PR, right?

Comment (rodrigozhou) [reply service/history/outbound_queue_standby_task_executor_test.go:1]: Ugh, will remove

File: common/tasks/dynamic_worker_pool_scheduler.go [hunk 2]
```diff
@@ -145,7 +173,31 @@ func (pool *DynamicWorkerPoolScheduler) dequeueLocked() (task Runnable, ok bool)
...
+	bufferSizeCounter.With(pool.metricsHandler).Record(int64(bufferSize))
+}
+
+func (pool *DynamicWorkerPoolScheduler) recordActiveWorkers(activeWorkers int) {
+	activeWorkersCounter.With(pool.metricsHandler).Record(int64(activeWorkers))
+}
+
+func (pool *DynamicWorkerPoolScheduler) exportMetricsWorker() {
+	defer pool.wg.Done()
+	timer := time.NewTicker(10 * time.Second)
```
[the remark this answers is not in the document]
Comment (rodrigozhou) [reply common/tasks/dynamic_worker_pool_scheduler.go:172]: Added a TODO

File: common/tasks/dynamic_worker_pool_scheduler.go (hunk 1 above)
Comment (bergundy) [reply common/tasks/dynamic_worker_pool_scheduler.go]: Is this coming in a separate PR?

**Review by bergundy — APPROVED** [review]

File: common/tasks/dynamic_worker_pool_scheduler.go (hunk 1 above)
Comment (rodrigozhou) [reply common/tasks/dynamic_worker_pool_scheduler.go]: I'll do it a separate PR.
